clients/ui/frontend/src/app/pages/modelCatalog/screens/RegisterCatalogModelForm.tsx

```diff
@@ -0,0 +1,243 @@
+import { ModelCatalogItem } from '~/app/modelCatalogTypes';
+import { ModelRegistry, ModelRegistryMetadataType } from '~/app/types';
+import { ModelRegistryContext } from '~/app/context/ModelRegistryContext';
+import useRegisteredModels from '~/app/hooks/useRegisteredModels';
+import { extractVersionTag } from '~/app/pages/modelCatalog/utils/modelCatalogUtils';
+import ModelRegistrySelector from '~/app/pages/modelRegistry/screens/ModelRegistrySelector';
+import { registeredModelUrl } from '~/app/pages/modelRegistry/screens/routeUtils';
+import { getCatalogModelDetailsRoute } from '~/app/routes/modelCatalog/catalogModelDetails';
+import {
+  catalogParamsToModelSourceProperties,
+  modelSourcePropertiesToCustomProperties,
+} from '~/concepts/modelRegistry/utils';
+
+interface RegisterCatalogModelFormProps {
+  model: ModelCatalogItem;
+  preferredModelRegistry: ModelRegistry;
+}
+
+const RegisterCatalogModelForm: React.FC<RegisterCatalogModelFormProps> = ({
+  model,
+  preferredModelRegistry,
+}) => {
+  const navigate = useNavigate();
+  const { apiState } = React.useContext(ModelRegistryContext);
+  const [registeredModels, registeredModelsLoaded] = useRegisteredModels();
+
+  const [isSubmitting, setIsSubmitting] = React.useState(false);
+  const [submitError, setSubmitError] = React.useState<Error | undefined>(undefined);
+
+  const versionTag = extractVersionTag(model.tags);
+
+  const catalogParams = {
+    modelName: model.name,
+    tag: versionTag || '',
+    sourceName: model.provider || '',
+    repositoryName: model.id || '',
+  };
+  const sourceProperties = catalogParamsToModelSourceProperties(catalogParams);
+  const sourceCustomProperties = modelSourcePropertiesToCustomProperties(sourceProperties);
+
+  const initialFormData: RegisterCatalogModelFormData = {
+    modelName: `${model.name}-${versionTag || ''}`,
+    modelDescription: model.description || '',
+    versionName: 'Version 1',
+    versionDescription: '',
+    sourceModelFormat: '',
+    sourceModelFormatVersion: '',
+    modelLocationType: ModelLocationType.URI,
+    modelLocationEndpoint: '',
+    modelLocationBucket: '',
+    modelLocationRegion: '',
+    modelLocationPath: '',
+    modelLocationURI: model.url || '',
+    modelRegistry: preferredModelRegistry.name,
+    modelCustomProperties: {},
+    versionCustomProperties: {
+      License: {
+        // eslint-disable-next-line camelcase
+        string_value: model.license || '',
+        metadataType: ModelRegistryMetadataType.STRING,
+      },
+      Framework: {
+        // eslint-disable-next-line camelcase
+        string_value: model.framework || '',
+        metadataType: ModelRegistryMetadataType.STRING,
+      },
+      Task: {
+        // eslint-disable-next-line camelcase
+        string_value: model.task || '',
+        metadataType: ModelRegistryMetadataType.STRING,
+      },
+    },
+    additionalArtifactProperties: sourceCustomProperties,
```

Comment:
The `additionalArtifactProperties` are the `sourceProperties`, not custom properties.